Server Panics On GET Moderators

This occurs when attempting to get a list of available moderators:

```
572315: 00000000000000000568954bb3147ecc9bf66aa72d8c000aa89e6274dc363db7
panic: interface conversion: routing.IpfsRouting is routinghelpers.Tiered, not *dht.IpfsDHT

goroutine 62811 [running]:
github.com/OpenBazaar/openbazaar-go/api.(*jsonAPIHandler).GETModerators.func3(0xc000748400, 0x59360a0, 0xc0000be008, 0xc001328d96, 0x7, 0xc002108da0, 0x16)
	/Users/jjeffryes/go/src/github.com/OpenBazaar/openbazaar-go/api/jsonapi.go:1841 +0x248
created by github.com/OpenBazaar/openbazaar-go/api.(*jsonAPIHandler).GETModerators
	/Users/jjeffryes/go/src/github.com/OpenBazaar/openbazaar-go/api/jsonapi.go:1840 +0xb4a
exit status 2
```

--- a/api/jsonapi.go
+++ b/api/jsonapi.go
@@ -9,12 +9,11 @@ import (
 	"encoding/json"
 	"fmt"
 
-	cid "gx/ipfs/QmPSQnBKM9g7BaUcZCvswUJVscQ1ipjmwxN5PXCjkp9EQ7/go-cid"
+	"gx/ipfs/QmPSQnBKM9g7BaUcZCvswUJVscQ1ipjmwxN5PXCjkp9EQ7/go-cid"
 	mh "gx/ipfs/QmPnFwZ2JXKnXgMw8CdBPxn7FWh6LLdjUjxV1fKHuJnkr8/go-multihash"
-	routing "gx/ipfs/QmPpYHPRGVpSJTkQDQDwTYZ1cYUR2NM4HS6M3iAXi8aoUa/go-libp2p-kad-dht"
-	peer "gx/ipfs/QmTRhk7cgjUf2gfQ3p2M9KPECNZEW9XUrmHcFCgog4cPgB/go-libp2p-peer"
+	"gx/ipfs/QmTRhk7cgjUf2gfQ3p2M9KPECNZEW9XUrmHcFCgog4cPgB/go-libp2p-peer"
 	ps "gx/ipfs/QmTTJcDL3gsnGDALjh2fDGg1onGRUdVgNL2hU2WEZcVrMX/go-libp2p-peerstore"
-	datastore "gx/ipfs/QmaRb5yNXKonhbkpNxNawoydk4N6es6b4fPj19sjEKsh5D/go-datastore"
+	"gx/ipfs/QmaRb5yNXKonhbkpNxNawoydk4N6es6b4fPj19sjEKsh5D/go-datastore"
 
 	"io/ioutil"
 	"net/http"
@@ -997,7 +996,7 @@ func (i *jsonAPIHandler) PATCHSettings(w http.ResponseWriter, r *http.Request) {
 func (i *jsonAPIHandler) GETClosestPeers(w http.ResponseWriter, r *http.Request) {
 	_, peerID := path.Split(r.URL.Path)
 	var peerIDs []string
-	peers, err := ipfs.Query(i.node.IpfsNode, peerID)
+	peers, err := ipfs.Query(i.node.DHT, peerID)
 	if err == nil {
 		for _, p := range peers {
 			peerIDs = append(peerIDs, p.Pretty())
@@ -1751,7 +1750,7 @@ func (i *jsonAPIHandler) GETModerators(w http.ResponseWriter, r *http.Request) {
 			}
 			return xs[:j]
 		}
-		peerInfoList, err := ipfs.FindPointers(i.node.IpfsNode.Routing.(*routing.IpfsDHT), ctx, core.ModeratorPointerID, 64)
+		peerInfoList, err := ipfs.FindPointers(i.node.DHT, ctx, core.ModeratorPointerID, 64)
 		if err != nil {
 			ErrorResponse(w, http.StatusInternalServerError, err.Error())
 			return
@@ -1838,7 +1837,7 @@ func (i *jsonAPIHandler) GETModerators(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusAccepted)
 		SanitizedResponse(w, string(respJSON))
 		go func() {
-			peerChan := ipfs.FindPointersAsync(i.node.IpfsNode.Routing.(*routing.IpfsDHT), ctx, core.ModeratorPointerID, 64)
+			peerChan := ipfs.FindPointersAsync(i.node.DHT, ctx, core.ModeratorPointerID, 64)
 
 			found := make(map[string]bool)
 			foundMu := sync.Mutex{}

--- a/ipfs/query.go
+++ b/ipfs/query.go
@@ -2,19 +2,11 @@ package ipfs
 
 import (
 	"context"
-	"errors"
-
 	routing "gx/ipfs/QmPpYHPRGVpSJTkQDQDwTYZ1cYUR2NM4HS6M3iAXi8aoUa/go-libp2p-kad-dht"
-	peer "gx/ipfs/QmTRhk7cgjUf2gfQ3p2M9KPECNZEW9XUrmHcFCgog4cPgB/go-libp2p-peer"
-
-	"github.com/ipfs/go-ipfs/core"
+	"gx/ipfs/QmTRhk7cgjUf2gfQ3p2M9KPECNZEW9XUrmHcFCgog4cPgB/go-libp2p-peer"
 )
 
-func Query(n *core.IpfsNode, peerID string) ([]peer.ID, error) {
-	dht, ok := n.Routing.(*routing.IpfsDHT)
-	if !ok {
-		return nil, errors.New("routing is not type IpfsDHT")
-	}
+func Query(dht *routing.IpfsDHT, peerID string) ([]peer.ID, error) {
 	id, err := peer.IDB58Decode(peerID)
 	if err != nil {
 		return nil, err

--- a/ipfs/query.go
+++ b/ipfs/query.go
@@ -2,28 +2,25 @@ package ipfs
 
 import (
 	"context"
+
 	routing "gx/ipfs/QmPpYHPRGVpSJTkQDQDwTYZ1cYUR2NM4HS6M3iAXi8aoUa/go-libp2p-kad-dht"
-	"gx/ipfs/QmTRhk7cgjUf2gfQ3p2M9KPECNZEW9XUrmHcFCgog4cPgB/go-libp2p-peer"
+	peer "gx/ipfs/QmTRhk7cgjUf2gfQ3p2M9KPECNZEW9XUrmHcFCgog4cPgB/go-libp2p-peer"
 )
 
+// Query returns the closest peers known for peerID
 func Query(dht *routing.IpfsDHT, peerID string) ([]peer.ID, error) {
 	id, err := peer.IDB58Decode(peerID)
 	if err != nil {
 		return nil, err
 	}
-
 	ch, err := dht.GetClosestPeers(context.Background(), string(id))
 	if err != nil {
 		return nil, err
 	}
+
 	var closestPeers []peer.ID
-	events := make(chan struct{})
-	go func() {
-		defer close(events)
-		for p := range ch {
-			closestPeers = append(closestPeers, p)
-		}
-	}()
-	<-events
+	for p := range ch {
+		closestPeers = append(closestPeers, p)
+	}
 	return closestPeers, nil
 }

